src/asami/index.cljc

```diff
@@ -8,29 +8,35 @@
             [zuko.logging :as log :include-macros true]
             [schema.core :as s :include-macros true]))
 
-(s/defn index-add :- {s/Any {s/Any #{s/Any}}}
-  "Add elements to a 3-level index"
-  [idx :- {s/Any {s/Any #{s/Any}}}
+(def Index {s/Any {s/Any {s/Any {(s/required-key :t) s/Int ;transaction id
+                                 (s/required-key :id) s/Int}}}}) ;statement id
+
+(s/defn index-add :- Index
+  "Add elements to a 4-level index.
+   If triple already exists, returns given index unchanged."
+  [idx :- Index
    a :- s/Any
    b :- s/Any
-   c :- s/Any]
+   c :- s/Any
+   t :- s/Int
```

Comment:
Could we please swap the positions of the t and id args, for consistency with the durable store?